### PR TITLE
Feature card name list

### PR DESCRIPTION
**Pull request by BazilMuralidharan**
URL for testing

- https://feature-cardNameList--hdfc-bank--aemsites.hlx.page/content/forms/af/hdfc

### CODE REVIEW

File: creditcards/corporate-creditcardFunctions.js [hunk 1]
```diff
@@ -493,19 +495,38 @@ const moveCCWizardView = (source, target) => {
...
+/**
+ * create a list of name to be dispayed on card dropdown in confirm card screen.
+ * @param {object} globals - globals variables object containing form configurations.
+ */
+const listNameOnCard = (globals) => {
+  const { firstName, middleName, lastName } = globals.functions.exportData();
+  const dropDownElementName = 'nameOnCardDropdown';
+  const pannelNameOnCard = globals.form.corporateCardWizardView.confirmCardPanel.cardBenefitsPanel.CorporatetImageAndNamePanel.nameOnCardDropdown;
+  const nameOnCard = formUtil(globals, pannelNameOnCard);
+  const initalLoadValue = setSelectOptions(composeNameOption(firstName, middleName, lastName), dropDownElementName);
```
Comment (vdua) [inline creditcards/corporate-creditcardFunctions.js:507]:
```
const options = composeNameOption(firstName, middleName, lastName)
const dropDownElement = globals.form.....nameOnCardDropDown
globals.functions.setProperty(dropDownElement,{enum: options, value: initialValue})
```

File: common/executeinterfaceutils.js [hunk 2]
```diff
@@ -203,7 +223,8 @@ const sendIpaRequest = (ipaRequestObj, globals) => {
         const { cardBenefitsTextBox } = globals.form.corporateCardWizardView.confirmCardPanel.cardBenefitsPanel.cardBenefitsFeaturesPanel;
         const cardBenefitsTextField = formUtil(globals, cardBenefitsTextBox);
         cardBenefitsTextField.setValue(response.productEligibility.productDetails[0].keyBenefits[0]);
-        moveWizardView('corporateCardWizardView', 'confirmCardPanel');
+        const { firsName, middleName, lastName } = currentFormContext.customerName;
```
Comment (kunal-adobe) [inline common/executeinterfaceutils.js:226]: Please declare variables in the function where it is required

File: creditcards/corporate-creditcardFunctions.js [hunk 3]
```diff
@@ -493,19 +495,38 @@ const moveCCWizardView = (source, target) => {
   navigateFrom?.dispatchEvent(event);
 };
 
+/**
+ * create a list of name to be dispayed on card dropdown in confirm card screen.
+ * @param {object} globals - globals variables object containing form configurations.
+ */
+const listNameOnCard = (globals) => {
+  const { firstName, middleName, lastName } = globals.functions.exportData();
```
[the remark this answers is not in the document]
Comment (BazilMuralidharan) [reply creditcards/corporate-creditcardFunctions.js]: Resolved

File: common/formutils.js [hunk 4]
```diff
@@ -140,6 +140,62 @@ const setDataAttributeOnClosestAncestor = (elementName, fieldValue, dataAttribut
...
+    combination(fn, ln); // firstName + lastName
+  }
+  if (mn && ln) {
+    combination(mn, ln); // middleName + lastName
+    combination(initial(mn), ln); // initial_middleName + lastName
+  }
+  const jsonObj = arrayObj?.map(JSON.stringify);
+  const uniqueSet = new Set(jsonObj);
+  const uniqueArrObj = Array.from(uniqueSet)?.map(JSON.parse);
+  return uniqueArrObj;
```
[the remark this answers is not in the document]
Comment (BazilMuralidharan) [reply common/formutils.js]: fixed it.

File: common/formutils.js [hunk 5]
```diff
@@ -140,6 +140,62 @@ const setDataAttributeOnClosestAncestor = (elementName, fieldValue, dataAttribut
...
+  optionLists?.forEach((option) => {
+    const optionElement = document.createElement('option');
+    optionElement.value = option?.value;
+    optionElement.textContent = option?.label;
+    const parent = selectOption?.parentNode;
+    selectOption?.appendChild(optionElement);
+    parent?.setAttribute('data-active', true);
+  });
+  const defSelctedValue = selectOption?.value;
+  return defSelctedValue;
```
[the remark this answers is not in the document]
Comment (BazilMuralidharan) [reply common/formutils.js]: fixed it.

File: creditcards/corporate-creditcardFunctions.js (hunk 1 above)
Comment (BazilMuralidharan) [reply creditcards/corporate-creditcardFunctions.js]: fixed it.

File: common/executeinterfaceutils.js (hunk 2 above)
Comment (BazilMuralidharan) [reply common/executeinterfaceutils.js]: fixed it.

File: creditcards/corporate-creditcardFunctions.js [hunk 6]
```diff
@@ -665,10 +638,14 @@ const checkUserProceedStatus = (panStatus, globals) => {
 
 /**
  * Creates a PAN validation request object and handles success and failure callbacks.
+ * @param {string} firstName - The first name of the cardholder.
+ * @param {string} middleName - The last name of the cardholder.
+ * @param {string} lastName - The last name of the cardholder.
  * @param {Object} globals - The global object containing necessary data for PAN validation.
  * @returns {Object} - The PAN validation request object.
  */
-const createPanValidationRequest = (firstName, middleName, lastName, globals) => {
```
[the remark this answers is not in the document]
Comment (BazilMuralidharan) [reply creditcards/corporate-creditcardFunctions.js:671]: fixed it.

**Review by srivastava21ashish — APPROVED** [review]

**Pull request merged**